packages/medusa/src/strategies/batch-jobs/price-list/import.ts

```diff
@@ -0,0 +1,545 @@
+/* eslint-disable valid-jsdoc */
+import { EntityManager } from "typeorm"
+import { MedusaError, computerizeAmount } from "medusa-core-utils"
+
+import { AbstractBatchJobStrategy, IFileService } from "../../../interfaces"
+import CsvParser from "../../../services/csv-parser"
+import {
+  BatchJobService,
+  ProductService,
+  ProductVariantService,
+  PriceListService,
+  RegionService,
+  SalesChannelService,
+  ShippingProfileService,
+} from "../../../services"
+import { CreateBatchJobInput } from "../../../types/batch-job"
+import {
+  InjectedProps,
+  OperationType,
+  PriceListImportOperation,
+  PriceListImportOperationPrice,
+  ParsedPriceListImportPrice,
+  PriceListImportBatchJob,
+  PriceListImportCsvSchema,
+  TBuiltPriceListImportLine,
+  TParsedPriceListImportRowData,
+} from "./types"
+import { FlagRouter } from "../../../utils/flag-router"
+
+/**
+ * Process this many variant rows before reporting progress.
+ */
+const BATCH_SIZE = 100
+
+/**
+ * Default strategy class used for a batch import of products/variants.
+ */
+class PriceListImportStrategy extends AbstractBatchJobStrategy {
+  static identifier = "price-list-import-strategy"
+
+  static batchType = "price-list-import"
+
+  private processedCounter: Record<string, number> = {}
+
+  protected readonly featureFlagRouter_: FlagRouter
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly fileService_: IFileService
+
+  protected readonly regionService_: RegionService
+  protected readonly priceListService_: PriceListService
+  protected readonly productService_: ProductService
+  protected readonly batchJobService_: BatchJobService
+  protected readonly salesChannelService_: SalesChannelService
+  protected readonly productVariantService_: ProductVariantService
+  protected readonly shippingProfileService_: ShippingProfileService
```

Comment:
suggestion: remove unused injected deps (salesChannelService_, shippingProfileService_)
```suggestion
  protected readonly regionService_: RegionService
  protected readonly priceListService_: PriceListService
  protected readonly productService_: ProductService
  protected readonly batchJobService_: BatchJobService
  protected readonly productVariantService_: ProductVariantService
```